dragonfly/block/fence_gate.go

```diff
@@ -0,0 +1,135 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/model"
+	"github.com/df-mc/dragonfly/dragonfly/block/wood"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/df-mc/dragonfly/dragonfly/world/sound"
+	"github.com/go-gl/mathgl/mgl64"
+)
+
+// FenceGate is a block that can be used as an openable 1x1 barrier.
+type FenceGate struct {
```

Comment:
Might be a good idea changing this to `WoodFenceGate`. I'm aware there currently only are wood fence gates, but it would save us some trouble if they do decide to introduce non-wood fence gates.